tests/xharness/Jenkins.cs

```diff
@@ -147,6 +147,137 @@ bool IsIncluded (TestProject project)
 			return true;
 		}
 
+		class TestData
+		{
+			public string Variation;
+			public string MTouchExtraArgs;
+			public bool Debug;
+			public bool Profiling;
+		}
+
+		IEnumerable<TestData> GetTestData (RunTestTask test)
+		{
+			switch (test.ProjectPlatform) {
+			case "iPhone":
+				/* we don't add --assembly-build-target=@all=staticobject because that's the default in all our test projects */
+				yield return new TestData { Variation = "AssemblyBuildTarget: dylib (debug)", MTouchExtraArgs = "--assembly-build-target=@all=dynamiclibrary", Debug = true, Profiling = false };
+				yield return new TestData { Variation = "AssemblyBuildTarget: SDK framework (debug)", MTouchExtraArgs = "--assembly-build-target=@sdk=framework=Xamarin.Sdk --assembly-build-target=@all=staticobject", Debug = true, Profiling = false };
+
+				yield return new TestData { Variation = "AssemblyBuildTarget: dylib (debug, profiling)", MTouchExtraArgs = "--assembly-build-target=@all=dynamiclibrary", Debug = true, Profiling = true };
+				yield return new TestData { Variation = "AssemblyBuildTarget: SDK framework (debug, profiling)", MTouchExtraArgs = "--assembly-build-target=@sdk=framework=Xamarin.Sdk --assembly-build-target=@all=staticobject", Debug = true, Profiling = true };
+
+				yield return new TestData { Variation = "Release", MTouchExtraArgs = "", Debug = false, Profiling = false };
+				yield return new TestData { Variation = "AssemblyBuildTarget: SDK framework (release)", MTouchExtraArgs = "--assembly-build-target=@sdk=framework=Xamarin.Sdk --assembly-build-target=@all=staticobject", Debug = false, Profiling = false };
+				break;
+			case "iPhoneSimulator":
+				switch (test.TestName) {
+				case "monotouch-test":
+					yield return new TestData { Variation = "Debug (static registrar)", MTouchExtraArgs = "--registrar:static", Debug = true, Profiling = false };
```

Comment:
I assume no entry is required for dynamic because it's the default ?

Reply:
Yes, exactly, this method returns additional configurations. I'll add a comment about it.